### DISCUSSION TITLE
Remove parseEther, parseGwei, parseWei

### DISCUSSION THREAD

**Discussion by greenlucid** [wevm/viem#126] [Idea / Feature Request]
Since this library appears to strive for minimalism, wouldn't it make sense to unify these methods onto `parseUnits` instead?

the respective section in the docs would go from:

```typescript
import { parseEther } from 'viem'

parseEther('1')
```

to:

```typescript
import { parseUnits } from 'viem'

parseUnits('1', 18)
```

## Rationale

When reading at a method name such as `parseEther`, I find myself asking question such as:

> did this mean `parseSomethingToEther`, or `parseEtherToSomething`?

> what is that something?

Whereas, if I see something like `parseUnits('1', 18)` I can immediately tell this means:

> take the input, and add 18 decimals.

I'm not sure if this is the answer, but, does anyone else think this is more readable, or would this suck in practice?

## Pet peeve on units

I don't love `ether`, `gwei`, `wei` as units. We're used to them at this point, but they're not very aptly named. `Ether` unit will often come in the context of counting another ERC20, and that can confuse anyone.

**Comment by mlshv** [comment]
I agree with 

> did this mean parseSomethingToEther, or parseEtherToSomething?
> what is that something?

But `parseUnits('1', 18)` also isn't clear in this sense. Personally, this is not obvious for me:

> take the input, and add 18 decimals.

**Reply by jxom** [reply]
Yeah, I can't say I am in love with the naming of `parseEther`, etc... but I do believe an abstraction is needed to guard folks from inserting the wrong units (which could lead to disasterous UX consequences).
